### DISCUSSION TITLE
web端引入libpag.wasm 文件体积过大 有什么解决办法么

### DISCUSSION THREAD

**Discussion by zxcv66666** [Tencent/libpag#1719] [Q&A] [answered]
如题 web端 使用 npm i的方式安装libpag ，然后 通过webpack copy-webpack-plugin 将该libpag.wasm 文件 复制到 dist包中，但是这个文件大小有3.1M，导致首屏下载的时候 耗费时间过长，又尝试了cdn引入的办法，但是总是加载失败。另外 在ios或安卓 移动端浏览器打开该web 页面 首屏 会有2-3s加载时间，怀疑也是因为要下载libpag.wasm 文件导致的，这个有什么优化办法么。还有一个问题是在安卓端有的复杂的动画会不能加载出来，这个是什么原因导致的，有什么可以优化的方式么

**Comment by domchen** [comment]
开启一下服务器的gzip压缩，这个文件实际加载过程只有900多K

**Reply by yangchongduo** [reply]
废话

**Reply by baiyang9-tal** [reply]
所以900多k是一个很小的体积，可以秒开是不是😊

**Reply by domchen** [reply, marked as answer]
web环境的限制并没有十全十美的方案。开启gzip确实是有效的降低方式，但gzip后的900k这个是全功能版本的极限大小，没办法再减小了。还有就是采用只支持BMP预合成的lite版本，那个只有几十K。再不适用就要考虑采用别的方案。

**Comment by 95sunny** [comment]
解决了吗， 我在考虑使用 Cos服务器存储， 但是没找到配置引用地址的 api

**Comment by WayneGongCN** [comment]
如果 wasm 体积无法再优化，是否可以开放 wasm 加载方式的自定义呢？

目前在 `libpag.PAGInit` 时只有一个 `locateFile` 的方法用来指定 wasm 路径

如何能够支持自行控制 wasm 加载方式/时机，这样业务就能提前将 wasm 预加载/缓存到 localStorage/IndexDB，然后在`libpag.PAGInit` 时传入 wasm 的 ArrayBuffer
